packages/iov-bns/src/bnsconnection.spec.ts

```diff
@@ -1064,6 +1110,158 @@ describe("BnsConnection", () => {
     });
   });
 
+  describe("liveTx", () => {
+    it("finds an existing transaction", async () => {
+      pendingWithoutBnsd();
+      const connection = await BnsConnection.establish(bnsdTendermintUrl);
+      const chainId = connection.chainId();
+
+      const { profile, mainWalletId, faucet } = await userProfileWithFaucet(chainId);
+
+      const memo = `Payment ${Math.random()}`;
+      const sendTx: SendTransaction = {
+        kind: "bcp/send",
+        creator: faucet,
+        recipient: await randomBnsAddress(),
+        memo: memo,
+        amount: defaultAmount,
+      };
+
+      const nonce = await connection.getNonce({ pubkey: faucet.pubkey });
+      const signed = await profile.signTransaction(mainWalletId, faucet, sendTx, bnsCodec, nonce);
+      const response = await connection.postTx(bnsCodec.bytesToPost(signed));
+      const transactionIdToSearch = response.transactionId;
+      await response.blockInfo.waitFor(info => info.state !== TransactionState.Pending);
+
+      await tendermintSearchIndexUpdated();
```

Comment:
Actually, this brings up an interesting point for me....

If we call `liveTx` 2 ms after the tx event was emitted, but 10 ms before the search index was updated, then it may get no event at all.

For true robustness, we should update blockInfo not after the Header event, but after the propsoed `IndexedBlock` event (of course this waits for some future version of tendermint, just wanted to write down idea as I was noticing it in code)